### PR TITLE
model storage blueprint

### PR DESCRIPTION
**Pull request by Xe**
Ref TIG-2783

Wasn't able to get Skypilot working, so I documented what I was using to test this.

- [x] Add blurb about Fly.io to fly.io specific page
- [x] Add blurb about vast.ai to vast.ai specific page

### CODE REVIEW

File: docs/blueprints/model-storage/index.md [hunk 1]
```diff
@@ -0,0 +1,315 @@
+# Storing Model Weights in Tigris
+
+The most common way to deploy AI models in production is by using “serverless”
+inference. This means that every time you get a request, you don’t know what
+state the underlying hardware is in. You don’t know if you have your models
+cached, and in the worst case you need to do a cold start and download your
+model weights from scratch.
+
+This is typically done with the Hugging Face CDN, but sometimes that’s just not
```
Comment (ovaistariq) [inline docs/blueprints/model-storage/index.md:9]: Even with Tigris you are storing your model weights with a third party, right? So probably this argument doesn't hold as much weight.

File: docs/blueprints/model-storage/index.md [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+live and die in a single part of the globe. Remember, while your instances are
+sitting there pulling model weights from the cloud, you're burning GPU spend.
+Time is money.
+
+## Usecase
+
+You can put AI model weights into Tigris so that they are cached and fast no
+matter where you’re inferencing from. This allows you to have cold starts be
+faster and you can take advantage of Tigris'
+[globally distributed pull-through caching architecture](/docs/overview/),
```
Comment (ovaistariq) [inline docs/blueprints/model-storage/index.md:21]: globally distributed architecture

File: docs/blueprints/model-storage/index.md [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+![The word 'success' in front of the Space Needle](./success.webp)
+
+You can destroy the machine with this command:
+
+```text
+fly machine destroy --force -a your-app-name-here fluxschnell
+```
+
+### Skypilot
```
Comment (ovaistariq) [inline docs/blueprints/model-storage/index.md:208]: Should we move this out to its own page about Skypilot?

File: docs/blueprints/model-storage/index.md [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+| ----------------------: | :----------------------------------------------------- |
+|     `AWS_ACCESS_KEY_ID` | The access key ID from the runner keypair              |
+| `AWS_SECRET_ACCESS_KEY` | The secret access key from the runner keypair          |
+|   `AWS_ENDPOINT_URL_S3` | `https://fly.storage.tigris.dev`                       |
+|            `AWS_REGION` | `auto`                                                 |
+|     `MODEL_BUCKET_NAME` | `model-storage` (replace with your own bucket name)    |
+|            `MODEL_PATH` | `black-forest-labs/FLUX.1-schnell`                     |
+|    `PUBLIC_BUCKET_NAME` | `generated-images` (replace with your own bucket name) |
+
+### fly.io
```
Comment (ovaistariq) [inline docs/blueprints/model-storage/index.md:130]: Should we move this out to its own doc page?

File: docs/blueprints/model-storage/index.md [hunk 5]
```diff
@@ -0,0 +1,315 @@
+# Storing Model Weights in Tigris
```
Comment (ovaistariq) [inline docs/blueprints/model-storage/index.md:1]: My main high level comment is whether the instructions here can be simplified even further

File: docs/blueprints/model-storage/index.md (hunk 1 above)
Comment (Xe) [reply docs/blueprints/model-storage/index.md:9]: I more meant that huggingface downtime won't impact your uptime. Maybe need to rephrase this.

File: docs/blueprints/model-storage/index.md (hunk 2 above)
Comment (Xe) [reply docs/blueprints/model-storage/index.md:21]:
```suggestion
[globally distributed architecture](/docs/overview/),
```

I was going to say pull-through caching architecture here to point to that auto globally replicated bit, but I can let that go.

File: docs/blueprints/model-storage/index.md (hunk 3 above)
Comment (Xe) [reply docs/blueprints/model-storage/index.md:208]: Possibly? I wanted to keep it all on one page to make things control-f-able.

File: docs/blueprints/model-storage/index.md (hunk 4 above)
Comment (Xe) [reply docs/blueprints/model-storage/index.md:130]: Possibly? I wanted to keep it all on one page to make things control-f-able.

Comment (Xe) [reply docs/blueprints/model-storage/index.md]: Implemented in https://github.com/tigrisdata/tigris-os-docs/pull/114/commits/650c637b5f64edc321c4cb81da2b40137e272a7f

File: docs/blueprints/model-storage/index.md (hunk 3 above)
Comment (Xe) [reply docs/blueprints/model-storage/index.md]: Implemented in https://github.com/tigrisdata/tigris-os-docs/pull/114/commits/650c637b5f64edc321c4cb81da2b40137e272a7f

File: docs/blueprints/model-storage/index.md (hunk 1 above)
Comment (katieschilling) [reply docs/blueprints/model-storage/index.md:9]: > This is typically done with the Hugging Face CDN, but sometimes that’s just not fast enough, you don’t want to distribute your private model weights over a third party, or compliance issues force you to make sure that your model weights live and die in a single part of the globe.

-->

> A couple fixable problems arise when running your models on serverless or any frequently changing infrastructure: 

- Model distribution that's not optimized for latency causes needless GPU idle time as the model weights are downloaded to the machine on cold start. Tigris behaves like a content delivery network by default and is designed for low latency, saving idle time on cold start.
- Compliance restrictions like data sovereignty and GDPR increase complexity quickly. Tigris makes regional restrictions a one-line configuration, guide [here](https://www.tigrisdata.com/docs/objects/object_regions/).
- Reliance on third party caches for distributing models creates an upstream dependency and leaves your system vulnerable to downtime. Tigris guarantees 99.99% availability with [publicly available data](https://www.tigrisdata.com/blog/availability-metrics-public/).

I'm sitting with the formatting as it seems busy but this is the argument imho

Comment (katieschilling) [reply docs/blueprints/model-storage/index.md:9]: And I wouldn't name huggingface directly in this context

**Review by katieschilling — APPROVED** [review]

**Pull request merged**